pkg/apis/apps/validation/validation.go

```diff
@@ -754,8 +759,13 @@ func ValidatePodTemplateSpecForReplicaSet(template *api.PodTemplateSpec, selecto
 			}
 		}
 		allErrs = append(allErrs, apivalidation.ValidatePodTemplateSpec(template, fldPath, opts)...)
-		if replicas > 1 {
-			allErrs = append(allErrs, apivalidation.ValidateReadOnlyPersistentDisks(template.Spec.Volumes, fldPath.Child("spec", "volumes"))...)
+		// Daemons run on more than one node, Cancel verification of read and write volumes.
+		var oldVols []api.Volume
+		if oldTemplate != nil {
+			oldVols = oldTemplate.Spec.Volumes
+			if replicas > 1 {
```

Comment:
This wants to be outside the `oldTemplate != nil` block I think?

Reply:
Maybe I didn't write it well :(